src/OpenTelemetry.AutoInstrumentation.StartupHook/RulesEngine/ApplicationInExcludeListRule.cs

```diff
@@ -30,19 +31,33 @@ public ApplicationInExcludeListRule()
 
     internal override bool Evaluate()
     {
-        var applicationName = GetApplicationName();
+        var appDomainName = GetAppDomainName();
+        var processModuleName = GetProcessModuleName();
 
-        if (IsApplicationInExcludeList(applicationName))
+        if (IsApplicationInExcludeList(appDomainName, processModuleName))
```

Comment:
In order for us to give the exact reason why the code bailed out it seems better to check each condition separately instead of having a method checking both at the same time. Then the code can log the exact reason why we bailed out, without requiring debug logging.

Otherwise, LGTM.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/2765/commits/3f7f31c306e93e0f5fabd0020d7d400c72cfa0d3